nntrainer/layers/split_layer.cpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file   split_layer.cpp
+ * @date   21 May 2021
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is Split Layer Class for Neural Network
+ *
+ */
+
+#include <cstring>
+#include <layer_internal.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <split_layer.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+const std::string SplitLayer::type = "split";
+
+int SplitLayer::initialize(Manager &manager) {
+  int status = ML_ERROR_NONE;
+
+  if (split_dimension < 1) {
+    ml_loge("Error: cannot split along the batch dimension");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (split_dimension >= MAXDIM) {
+    ml_loge("Error: split dimension exceeding the total number of dimensions");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (getNumInputs() == 0) {
+    ml_loge("Error: number of inputs are not initialized");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (getNumInputs() > 1) {
+    ml_loge("Error: only a single input is supported with split layer");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  const TensorDim &in_dim = input_dim[0];
+  setNumOutputs(in_dim.getTensorDim(split_dimension));
```

Comment:
Could you add comments here? how exactly we expect the split_dimension from user? batch size must be same, and height and width??

Reply:
I will add.

Reply:
done.